deku-derive/src/macros/deku_read.rs

```diff
@@ -220,6 +220,16 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
             let initialize_enum =
                 super::gen_enum_init(variant_is_named, variant_ident, internal_fields);
 
+            if !has_discriminant && variant.id_pat.is_none() {
```

Comment:
It might be possible to do this section earlier when we're figuring out the variant id:
https://github.com/sharksforarms/deku/blob/afa7672dbf86977a93eda5ed874d0bc77aa4171d/deku-derive/src/macros/deku_read.rs#L179-L183

Reply:
Added Option return for the let assignment @ 179, I think it's much nicer.

Reply:
Agreed!